sklearn/preprocessing/data.py

```diff
@@ -2028,7 +2028,11 @@ def fit(self, X, y=None):
 
         if y is not None:
             for i in range(len_data):
-                X_key = tuple(X.take(i, axis=0).take(inclusion_used).flatten())
+                X_key_1 = X.take(i, axis=0)
```

Comment:
I plan on using A.take([i], axis=0) to handle edge cases where the array A is a single 1D array `[a_1, ..., a_n]`  
Because if I just do A[i], and A[i] is a scalar instead of a numpy array, then it will throw an error when I try to do things like len(A[i])  
Whereas if I do A.take([i], axis=0), and if A is a 1D array, then the result will just be a 1x1 array and I can still do len() on it.  
Would this be the correct approach on handling such cases?